### PR TITLE
EASY-2322 POC. Split up mixed lib/rpm project.

### PR DESCRIPTION
**Pull request by janvanmansum**
@DANS-KNAW/easy 

- [ ] Don't merge, yet. This is a POC. `easy-ingest-flow`'s dependencies will have to be updated if we are going to do this.

This is a POC for EASY-2322. It splits up `easy-archive-bag` into a library-project (which builds a simple jar containing most of the code) and a command-project (which builds the command line executable, installable with RPM, and which used the lib to implement its functionality).

#### When applied it will
* Convert this project into a multi-module Maven project with 2 modules: `lib` and `command`
* If you build the complete project it will deliver two artifacts `easy-archive-bag-lib` and `easy-archive-bag` (the command)
* If you deploy the complete project it will publish the lib to maven-snapshots and the RPM to rpm-snapshots

#### Where should the reviewer @DANS-KNAW/easy start?
* Fetch the branch and check it out locally to examine the new project structure.

#### How should this be manually tested?
* Build the multi-module project
* If you have credentials for Nexus, deploy it.

### CODE REVIEW

File: command/src/main/scala/nl.knaw.dans.easy.archivebag/Command.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.archivebag
+
+import nl.knaw.dans.lib.error._
+
+object Command extends {
```
Comment (rvanheest) [inline command/src/main/scala/nl.knaw.dans.easy.archivebag/Command.scala]: either do `extends App` and remove `def main` or remove `extends`

File: command/pom.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            <releases>
+                <enabled>false</enabled>
+            </releases>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+            <url>https://maven.dans.knaw.nl/snapshots/</url>
+        </repository>
+    </repositories>
+    <profiles>
```
Comment (rvanheest) [inline command/pom.xml]: don't we get this part in `dans-scala-app-project` or any of its parent projects?
same for `repositories` above...

File: command/src/main/scala/nl.knaw.dans.easy.archivebag/Command.scala (hunk 1 above)
Comment (janvanmansum) [reply command/src/main/scala/nl.knaw.dans.easy.archivebag/Command.scala]: Strange that this compiles, by the way.

File: command/pom.xml (hunk 2 above)
Comment (janvanmansum) [reply command/pom.xml]: Chicken-and-egg problem. How do we download `dans-scala-app-project` then, if it is not already in our local Maven cache?

**Comment by rvanheest** [comment]
I've been thinking about this a bit more, especially both Maven projects having the same package name. Have you looked into what limitations this has? For example, can now both the `lib` and `app` parts have a class with the same name (probably not) and how do `package.scala` files behave in this situation.
Perhaps we should consider for these projects to have a `nl.knaw.dans.easy.<name>.lib` and a `nl.knaw.dans.easy.<name>.app` package, both in their respective Maven projects.

**Comment by janvanmansum** [comment]
> I've been thinking about this a bit more, especially both Maven projects having the same package name. Have you looked into what limitations this has? For example, can now both the `lib` and `app` parts have a class with the same name (probably not) and how do `package.scala` files behave in this situation.
> Perhaps we should consider for these projects to have a `nl.knaw.dans.easy.<name>.lib` and a `nl.knaw.dans.easy.<name>.app` package, both in their respective Maven projects.

Yes, I think that would be a good idea.

**Comment by janvanmansum** [comment]
Proof of concept finished. Closing PR. In the JIRA issue I have modified @rvanheest's idea a bit. Instead of `<name>.app` and `<name>.lib` I propose keeping the lib-code in the same package (this is after all the main logic) and put the app-code in `<name>.command` (its only function is to provide the ability to execute the code as a command).

**Pull request closed without merging**